### DISCUSSION TITLE
KAFKA SINK encoding error

### DISCUSSION THREAD

**Discussion by coredump17** [vectordotdev/vector#18912] [General]
Hi All, I am trying to receive logs from syslog and forward to kafka.

Oct 23 18:04:51 xxxvector[594955]: 2023-10-23T17:04:51.247550Z DEBUG vector::app: messaged="Building runtime." worker_threads=80
Oct 23 18:04:51 xxxvector[594955]: 2023-10-23T17:04:51.303469Z  INFO vector::app: Loading configs. paths=["/conf/config.json"]
Oct 23 18:04:51 xxxvector[594955]: 2023-10-23T17:04:51.305713Z ERROR vector::cli: Configuration error. error=missing field `encoding'

my  configuration:

  "sinks": {
    "to_kafka": {
      "type": "kafka",
      "inputs": [
        "json_src"
      ],
      "bootstrap_servers": "10.14.22.123:9092,10.14.23.332:9092",
      "topic": "test",
      "encoding.codec": "raw_message",
      "compression": "none",
      "key_field": "hostname",
      "librdkafka_options": {
        "client.id": "xxx",
        "ssl.key.location": "config/certs/cxxx.key",
        "ssl.certificate.location": "config/certs/xxx.crt",
        "ssl.ca.location": "config/certs/truststore.pem"
      }
    }
  }
}

What should the encoding be? i tried encoding.codec json and raw message but it will wants 'encoding' key. 

thanks in advance!

**Comment by jszwedko** [comment]
I think you need:

```
"encoding": {
  "codec": "raw_message"
}
```

That is, `encoding` should be an object.

**Comment by coredump17** [comment]
How silly do I feel. Many thanks
